code/game/machinery/slotmachine.dm

```diff
@@ -293,16 +325,21 @@
 	return amount
 
 /obj/machinery/computer/slot_machine/proc/dispense(amount = 0, cointype = /obj/item/coin/silver, mob/living/target, throwit = 0)
-	var/value = coinvalues["[cointype]"]
-
-
-	while(amount >= value)
-		var/obj/item/coin/C = new cointype(loc) //DOUBLE THE PAIN
-		amount -= value
+	if(paymode == HOLOCHIP)
+		var/obj/item/holochip/H = new /obj/item/holochip(loc,amount)
+		
 		if(throwit && target)
-			C.throw_at(target, 3, 10)
-		else
-			random_step(C, 2, 40)
+			H.throw_at(target, 3, 10)
+	else
+		var/value = coinvalues["[cointype]"]
+
+		while(amount >= value)
+			var/obj/item/coin/C = new cointype(loc) //DOUBLE THE PAIN
+			amount -= value
+			if(throwit && target)
+				C.throw_at(target, 3, 10)
+			else
+				random_step(C, 2, 40)
```

Comment:
Interestingly enough, this block means you can make more than what you're suppose to get if the amount of the coin doesn't divide evenly with the value of the `cointype`

Reply:
I think this happened to me via testing. I remember having around 95 coins in balance but when I took them out and put back I had 100. I can try to refactor these, since it can only spit silver and iron coins(Silver is 10, iron is 1), but I am not sure what the emagged state is. If is emagged, it only gives silver coins?

Reply:
`while((amount-value) >= value` but that would give you less than the targeted amount if it doesn't divide evenly.

nbd honestly we can just chalk it up to luck bonus lol